warplib/maths/bytes_access.cairo

```diff
@@ -0,0 +1,85 @@
+from warplib.maths.pow2 import pow2
+from starkware.cairo.common.bitwise import bitwise_and
+from warplib.maths.utils import narrow_safe
+
+from starkware.cairo.common.cairo_builtins import BitwiseBuiltin
+from starkware.cairo.common.uint256 import Uint256
+from starkware.cairo.common.math import assert_le_felt
+from starkware.cairo.common.math_cmp import is_le_felt
+
+from starkware.cairo.common.serialize import serialize_word
+
+func byte_accessor{range_check_ptr}(index : felt) -> (offset : felt):
+    assert_le_felt(index, 31)
+    let (pow2index) = pow2(index * 8)
+    if index == 31:
```

Comment:
With the changes suggested below, the highest valid index into a felt is 30 (as that's the highest valid index into a uint248 and for uint256s you can index into each felt, and each up to a maximum of 16), so it doesn't make sense to have a different return here. There should be an assert or just no check if the asserts in the other functions are sufficient